### ISSUE TITLE
PHP Warning after update to 2.1.19

### ISSUE THREAD

**Issue by flack** [Studio-42/elFinder#1800]
I've recently updated from 2.1.16 to 2.1.19 and since then, I get this error sporadically in JSON responses:

```
Warning: session_start(): Cannot send session cookie - headers already sent by (output started at vendor/studio-42/elfinder/php/elFinderConnector.class.php:317) in vendor/studio-42/elfinder/php/elFinderSession.php on line 47
```

EDIT: The last working version for me was actually 2.1.16, I've corrected the description above. This only happens on a PHP 5.3 server currently, I can't reproduce unter PHP 5.6

**Comment by nao-pon** [comment]
@flack I'm checking on PHP 5.3.29 server. But I can't reproduce of this problem yet.

I tried next steps.

+ Upload elFider 2.1.19 to PHP 5.3.29 server
+ Rename connector.minimal.php.dist to connector.minimal.php
+ Make new file
+ Upload JPEG image file
+ Dupulicate image file
+ Remove image file
+ Edit text file

That's no problem.

**Comment by flack** [comment]
I'm on PHP 5.3.3 (Debian 6) and I'm using a custom adapter (extends `elFinderVolumeDriver`, source is available here: https://github.com/flack/openpsa/blob/master/lib/org/openpsa/documents/elFinderVolumeOpenpsa.php), so maybe this has something to do with it. But it worked fine since at least 2.1.8, until 2.1.16 (2.1.17 doesn't run at all on PHP 5.3, so I don't know if this would also exhibit the problem)

**Comment by nao-pon** [comment]
It's in not clear but I'm guessing that registered `midcom::get()->finish();` in the shutdown function may have some influence.

**Comment by flack** [comment]
Thanks for looking into the issue!

I've tried removing the `midcom::get()->finish();` line, and also completely removing the shutdown function, but the warning still appears

**Comment by flack** [comment]
I've bisected the issue now, and it looks like it comes from this commit: 7b8053a3bac1b3602830a04ade5a1f0150b708c1

**Comment by flack** [comment]
I've tried setting `noSessionCache` for the volume, but it didn't help. The problem goes away however when I remove this line: https://github.com/Studio-42/elFinder/blob/master/php/elFinderVolumeDriver.class.php#L675

**Comment by nao-pon** [comment]
What about next?

elFinderVolumeDriver.class.php
```php
	public function __destruct() {
		if (! headers_sent()) {
			$this->session->set($this->id, $this->sessionCache);
		}
	}
```

**Comment by flack** [comment]
Yes, that gets rid of the warning. BTW, I've looked a bit around, and I think this may only on PHP 5.3 because of this line:

https://github.com/Studio-42/elFinder/blob/master/php/elFinderSession.php#L41

Since I instantiate `elFinder` from inside my framework, there is already a session running (based on Symfony HTTPFoundation). So my guess would be that in php 5.4+, elFinder detects the running session and doesn't try to start, while on 5.3, `session_start()` gets called a prints the warning. So another possible workaround would be to change this line:

https://github.com/Studio-42/elFinder/blob/master/php/elFinderSession.php#L47

to look like this:

```php
    @session_start();
```

**Comment by flack** [comment]
I think I've found a cleaner solution now than suppressing error output and sent a pull request

**Comment by nao-pon** [comment]
@flack I fixed this issue (maybe) @fc1153d243555d1bfdac0df82f59ecee4c5d8bc9. Please reopen if you having this issue yet with [nightly build](https://github.com/Studio-42/elFinder/archive/2.1.zip).

**Comment by flack** [comment]
@nao-pon looks good!
